src/main/jbake/content/documentation/bundles/models.md

```diff
@@ -660,86 +612,78 @@ If you want to generate a bundle header compliant with Sling Models < 1.3.4 (i.e
         </instructions>
     </configuration>
 
-# Caching
 
-By default, Sling Models do not do any caching of the adaptation result and every request for a model class will
-result in a new instance of the model class. However, there are two notable cases when the adaptation result can be cached. The first case is when the adaptable extends the `SlingAdaptable` base class. Most significantly, this is the case for many `Resource` adaptables as `AbstractResource` extends `SlingAdaptable`.  `SlingAdaptable` implements a caching mechanism such that multiple invocations of `adaptTo()` will return the same object. For example:
 
-    ::java
-    // assume that resource is an instance of some subclass of AbstractResource
-    ModelClass object1 = resource.adaptTo(ModelClass.class); // creates new instance of ModelClass
-    ModelClass object2 = resource.adaptTo(ModelClass.class); // SlingAdaptable returns the cached instance
-    assert object1 == object2;
 
-While this is true for `AbstractResource` subclasses, it is notably **not** the case for `SlingHttpServletRequest` as this class does not extend `SlingAdaptable`. So:
 
-    ::java
-    // assume that request is some SlingHttpServletRequest object
-    ModelClass object1 = request.adaptTo(ModelClass.class); // creates new instance of ModelClass
-    ModelClass object2 = request.adaptTo(ModelClass.class); // creates another new instance of ModelClass
-    assert object1 != object2;
+# Discouraged annotations
 
-Since API version 1.3.4, Sling Models *can* cache an adaptation result, regardless of the adaptable by specifying `cache = true` on the `@Model` annotation.
+In earlier versions of Sling Models the use of the annotation `@Inject` was suggested and documented; but over time it turned out that it had 2 major issues:
 
+* This injection iterated through all available injectors and injected the first non-null value provided by an injector. This lead to unpredictable behavior, although the order is well-defined. The addition of the `@Source` annotation did not really help.
+* This iteration turned out to the performance bottleneck, especially if (optional) injections were not succesful, and then all other injectors have to be tried.
```

Comment:
turned out to be a performance bottleneck

Reply:
+1